service/frontend/http_api_server.go

```diff
@@ -318,6 +324,28 @@ func (h *HTTPAPIServer) incomingHeaderMatcher(headerName string) (string, bool)
 	return runtime.DefaultHeaderMatcher(headerName)
 }
 
+func (h *HTTPAPIServer) registerOpenAPIDocsHandlers() {
+	serve := func(version int, spec []byte) func(http.ResponseWriter, *http.Request, map[string]string) {
+		return func(w http.ResponseWriter, _ *http.Request, _ map[string]string) {
+			rdr, err := gzip.NewReader(bytes.NewReader(spec))
+			if err != nil {
+				h.logger.Error("failed to initialize openapi spec reader", tag.NewInt("version", version), tag.Error(err))
+				w.WriteHeader(http.StatusInternalServerError)
+				return
+			}
+			if _, err := io.Copy(w, rdr); err != nil {
```

Comment:
should we set a content-type header above here?

Reply:
Done!